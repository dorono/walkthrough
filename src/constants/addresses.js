export const ADDRESSES = {
    PEGNET_BURN_ADDRESS: 'EC2BURNFCT2PEGNETooo1oooo1oooo1oooo1oooo1oooo19wthin',
    IS_PRIVATE_KEY: 'IS_PRIVATE_KEY',
    IS_KEY: 'IS_KEY',
    IS_PROBABLY_A_KEY: 'IS_PROBABLY_A_KEY',
    IS_NOT_A_PRIVATE_ADDRESS: 'IS_NOT_A_PRIVATE_ADDRESS',
};

Object.freeze(ADDRESSES);
