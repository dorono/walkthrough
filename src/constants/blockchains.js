export const AVAILABLE_BLOCKCHAINS = {
    PUBLIC: {
        label: CONFIG.blockchainNetwork,
        value: 0,
        serviceName: 'mainnet',
        url: CONFIG.gatewayUrls.publicNet,
    },
    SHARED: {
        label: 'Shared Sandbox',
        value: 1,
        serviceName: 'shared',
        url: CONFIG.gatewayUrls.sharedSandbox,
    },
    PRIVATE: {
        label: 'Private Network',
        value: 2,
        serviceName: 'private_network',
        url: null,
    },
};

Object.freeze(AVAILABLE_BLOCKCHAINS);
