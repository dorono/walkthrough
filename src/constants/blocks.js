export const BLOCKS = {
    DIRECTORY: 'dblock', // Directory Block
    FACTOID: 'fblock', // Factoid Block
    ENTRY: 'eblock', // Entry Block
    PUBLIC_NETWORK: 'fnblock', // Public Factom Network
    BITCOIN_NETWORK: 'bnblock', // Cripto Block (Bitcoin and Ethereum)
    ETHEREUM_NETWORK: 'enblock', // Cripto Block (Bitcoin and Ethereum)
    DEFAULT: 'block', // Default Block
};

Object.freeze(BLOCKS);
