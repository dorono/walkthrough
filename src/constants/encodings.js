export const ENCODINGS = {
    FORMAT: {
        HEX: 'hex',
        RAW: 'raw',
        BASE64: 'base64',
        JSON: 'json',
    },
};

Object.freeze(ENCODINGS);
