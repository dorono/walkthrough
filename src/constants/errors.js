export const ERRORS = {
    CREDENTIALS: 'credentials',
    CUSTOM_CREDENTIALS_NOT_ALLOWED: 'custom-credentials-not-allowed',
    NETWORK: 'network',
    OTHER: 'other',
};

export const ERROR_MESSAGES = {
    AUTHENTICATION_FAILED: 'Authentication failed',
};

Object.freeze(ERRORS);
