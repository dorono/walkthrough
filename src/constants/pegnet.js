// these array items must be kept in the same order
export const PEGNET_TRANSACTION_TYPES = [
    'transfer',
    'conversion',
    'coinbase',
    'burn',
];
