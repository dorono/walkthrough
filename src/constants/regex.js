/*
  REGEX Behavior:
  RAW: check for chars forbidden by XML 1.0 specifications, plus the unicode replacement character U+FFFD
*/
export const REGEX = {
    VALIDATE: {
        RAW: /[\x00-\x08\x0E-\x1F\x7F-\uFFFF]/g,
        NOT_PRINTABLE: /[\x00-\x1F]/,
        SHA_256: /[A-Fa-f0-9]{64}/,
    },
    JSON: /("(\\u[a-zA-Z0-9]{4}|\\[^u]|[^\\"])*"(\s*:)?|\b(true|false|null)\b|-?\d+(?:\.\d*)?(?:[eE][+\-]?\d+)?)/g,
    FORMATTING_CHARS: /\n|\t|\r|\0/g,
};

Object.freeze(REGEX);
