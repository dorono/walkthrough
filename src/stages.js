export const STAGE_FACTOM = 'factom';
export const STAGE_ANCHORED = 'anchored';
export const STAGE_CONFIRMED = `${STAGE_FACTOM},${STAGE_ANCHORED}`;
export const STAGE_PENDING = 'replicated';

export const STAGE_PENDING_CHAIN_TEXT = 'This chain has not yet been written to the Factom Blockchain.';
export const STAGE_PENDING_ENTRY_TEXT = 'This entry has not yet been written to the Factom Blockchain.';
export const STAGE_PENDING_DATE = 'PENDING';

export const STAGE_NOT_AVAILABLE = 'Not yet available';
