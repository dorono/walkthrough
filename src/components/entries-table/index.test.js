import React from 'react';
import {shallow} from 'enzyme';
import {STAGE_PENDING, STAGE_CONFIRMED} from 'constants/stages';

import {EntriesTable} from './index';

const defaultProps = {
    contentColumnName: 'HASH',
    hasLink: true,
    data: [{
        stage: STAGE_CONFIRMED,
    }, {
        stage: STAGE_CONFIRMED,
    }],
    renderContent: jest.fn,
};

jest.mock('../../utils/date', () => {
    return {
        currentTimezone: () => '11/11/11',
        formatDateLong: () => '11111111',
    };
});

describe('Entries Table Component', () => {
    it('should render without errors', () => {
        shallow(<EntriesTable {...defaultProps} />);
    });

    it('should match snapshot', () => {
        expect(shallow(<EntriesTable {...defaultProps} />)).toMatchSnapshot();
    });

    it('should NOT display the <PendingLegend /> component if there are no pending entries', () => {
        const wrapper = shallow(<EntriesTable {...defaultProps} />);
        expect(wrapper.state('hasPendingEntries')).toBe(false);
        expect(shallow(<EntriesTable {...defaultProps} />)).toMatchSnapshot();
    });

    it(
        'should change the local state to display the <PendingLegend /> component if one or more entries are pending',
    () => {
        defaultProps.data = [{
            stage: STAGE_PENDING,
        }, {
            stage: STAGE_CONFIRMED,
        }];

        const wrapper = shallow(<EntriesTable {...defaultProps} />);
        expect(wrapper.state('hasPendingEntries')).toBe(true);
        expect(shallow(<EntriesTable {...defaultProps} />)).toMatchSnapshot();
    });
});
