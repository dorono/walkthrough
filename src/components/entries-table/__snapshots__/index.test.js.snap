// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entries Table Component should NOT display the <PendingLegend /> component if there are no pending entries 1`] = `
<Container
  showFullWidthBanner={false}
  title="Entries"
>
  <withRouter(Table)
    columns={
      Array [
        "CREATED (11/11/11)",
        "HASH",
      ]
    }
    ellipsis={
      Array [
        1,
      ]
    }
    interactive={true}
    leftPadding={0}
    rows={
      Array [
        Object {
          "stage": "factom,anchored",
        },
        Object {
          "stage": "factom,anchored",
        },
      ]
    }
    type="secondary"
  >
    <Component />
  </withRouter(Table)>
  <PendingLegend
    fullWidthBannerText="This entry has not yet been written to the Factom Blockchain."
    highlightBanner={false}
    show={false}
  />
</Container>
`;

exports[`Entries Table Component should change the local state to display the <PendingLegend /> component if one or more entries are pending 1`] = `
<Container
  showFullWidthBanner={false}
  title="Entries"
>
  <withRouter(Table)
    columns={
      Array [
        "CREATED (11/11/11)",
        "HASH",
      ]
    }
    ellipsis={
      Array [
        1,
      ]
    }
    interactive={true}
    leftPadding={0}
    rows={
      Array [
        Object {
          "stage": "replicated",
        },
        Object {
          "stage": "factom,anchored",
        },
      ]
    }
    type="secondary"
  >
    <Component />
  </withRouter(Table)>
  <PendingLegend
    fullWidthBannerText="This entry has not yet been written to the Factom Blockchain."
    highlightBanner={false}
    show={true}
  />
</Container>
`;

exports[`Entries Table Component should match snapshot 1`] = `
<Container
  showFullWidthBanner={false}
  title="Entries"
>
  <withRouter(Table)
    columns={
      Array [
        "CREATED (11/11/11)",
        "HASH",
      ]
    }
    ellipsis={
      Array [
        1,
      ]
    }
    interactive={true}
    leftPadding={0}
    rows={
      Array [
        Object {
          "stage": "factom,anchored",
        },
        Object {
          "stage": "factom,anchored",
        },
      ]
    }
    type="secondary"
  >
    <Component />
  </withRouter(Table)>
  <PendingLegend
    fullWidthBannerText="This entry has not yet been written to the Factom Blockchain."
    highlightBanner={false}
    show={false}
  />
</Container>
`;
